spec/reek/smells/control_parameter_spec.rb

```diff
@@ -270,7 +270,7 @@ def things(arg)
         end
       EOS
       ctx = Reek::Context::MethodContext.new(nil, Reek::Source::SourceCode.from(src).syntax_tree)
-      smells = detector.examine(ctx)
+      smells = detector.run_for(ctx)
```

Comment:
Now that I look at this, it seems rather odd that `#run_for` returns detected smells, but we also call `#report_on` to have the smells passed to the reporter. Perhaps removing that indirection will be something to do in a next refactoring.